### DISCUSSION TITLE
Payload timestamp is an object rather than a Unix timestamp

### DISCUSSION THREAD

**Discussion by jfaigan** [thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus#30] [Q&A]
According to the Sparkplug spec, the timestamp should be an unsigned 64 bit int
![image](https://github.com/thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus/assets/14929031/1982a785-38fe-4e4e-afcb-1f2a7d102de7)

In node-red, the value for timestamp seems to be an object
![image](https://github.com/thomassorensen2650/node-red-contrib-mqtt-sparkplug-plus/assets/14929031/07adf8c2-4f53-486f-b58a-c44f35081282)

Is there a way for me to get the actual timestamp from this object?

Thank you in advance!

**Comment by thomassorensen2650** [comment]
As far as I can remember this is how the protobuf javascript library is decoding 64Bit numbers using some 3rd party library, because javascript does (or did?) not have native 64bit integers. we should convert it back to timestamps when we decode. Where do you see this number?

**Comment by thomassorensen2650** [comment]
I did some investigation. The Sparkplug-payload package that we use to encode the data had a minor version update that introduced the behavior.

It's encoding all Uint64 as Long which is technically correct, but its something that we should handle in the node. This has been addressed in #40 

Will be included in next version.
